source/controllers/controller.py

```diff
@@ -0,0 +1,11 @@
+CONTROLLERS = {}
+
+def controller(code):
+    def register_controller(func):
+        CONTROLLERS[code] = func
+        return func
+
+    return register_controller
+
+def get_controller_func(controller):
+    return CONTROLLERS[controller] if controller in CONTROLLERS else None
```

Comment:
Consider throwing a ValueError. That way, the flow can be much simpler, and the exception will be caught (there is already a handler for ValueError that send a bad message response).